File: src/game/model/bomber-industry.ts
import { GameConstants } from '../game-constants.ts';
import { Bomber } from './bomber.ts';
import { ShippingIndustry } from './shipping-industry.ts';
import { Position } from './position.ts';
import { Ship } from './ship.ts';

export class BomberIndustry extends ShippingIndustry {
  getMaterialCost() { return GameConstants.BomberIndustry.MaterialCost; }
  protected createShip(position: Position): Ship {
    this.productionDuration = GameConstants.BomberIndustry.ProductionDuration;
    this.timeToCompletion = this.productionDuration;
    return new Bomber(position);
  }
}

File: app/src/main/java/space/model/BomberIndustry.java
package space.model;


public class BomberIndustry extends ShippingIndustry {

	public BomberIndustry(ShipCollection allies, PilotCollection pilots) {
		super(allies, pilots);
	}
	
	protected Ship createShip(Position planet) {
		timeToCompletion = 10.0f;
		return new Bomber(planet);
	}

}
